media-tv/tvbrowser-bin/tvbrowser-bin-4.2.7.ebuild

```diff
@@ -0,0 +1,39 @@
+# Copyright 2020-2022 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI="8"
+
+inherit desktop
+
+DESCRIPTION="Themeable and easy to use TV Guide - written in Java"
+HOMEPAGE="https://www.tvbrowser.org"
+MY_PN="tvbrowser"
+SRC_URI="mirror://sourceforge/project/tvbrowser/TV-Browser%20Releases%20%28Java%2011%20and%20higher%29/${PV}/${MY_PN}_${PV}_bin.tar.gz -> ${P}.tar.gz"
+LICENSE="GPL-3"
+SLOT="0"
+KEYWORDS="~amd64"
+
+JAVA_SLOT=11
+RDEPEND="|| ( dev-java/openjdk:${JAVA_SLOT}
```

Comment:
Does it work if you write all this as `RDEPEND="virtual/jre:${JAVA_SLOT}"`?

Reply:
I did not put `>=`, only replaced the `|| ( ... )` with a shorter (and hopefully more standard) virtual.  I was asking if there was a reason for writing it the long way, like something not working right with `virtual/jre`.

Reply:
I think previously virtual/xxx did not work because jdk 11 was not properly supported some time ago.

Also, the launcher file depends on the openjdk paths:
POSSIBLE_JAVA_PROGRAM_DIRS=(
    /usr/lib*/openjdk-${JAVA_SLOT}
    /opt/openjdk-bin-${JAVA_SLOT}
    /opt/openjdk-jre-bin-${JAVA_SLOT}
)

Reply:
How would I properly do this staying with the virtual/jre?

Reply:
The paths do not change.  Try `cat virtual/jre/jre-11-r2.ebuild`, cat `virtual/jdk/jdk-11-r2.ebuild`, see what they do.

Reply:
I have reinstated the old version until someone can tell me how to fix this.

Reply:
I can make it depend on virtual/jre, but if some other jdk variant gets added here in the future, the launcher will not find its java executable?

Reply:
That is, the user *has* to install openjdk for the launcher (and maybe even the app) to work.